classes/migration/upgrade/v3_6_0/I10404_UpdateCategoryImageNameFields.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_6_0/I10404_UpdateCategoryImageNameFields.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I10404_UpdateCategoryImageNameFields
+ *
+ * @brief Migration to update Category image data properties for compatibility with FieldUploadImage component
+ */
+
+namespace PKP\migration\upgrade\v3_6_0;
+
+use APP\file\PublicFileManager;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Enumerable;
+use Illuminate\Support\Facades\DB;
+use PKP\file\ContextFileManager;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I10404_UpdateCategoryImageNameFields extends Migration
+{
+    public const CHUNK_SIZE = 1000;
+
+    public function up(): void
+    {
+        $contextIds = app()->get('context')->getIds();
+        $fileNamesToMove = [];
+
+        foreach ($contextIds as $contextId) {
+            DB::table('categories')
+                ->where('context_id', '=', $contextId)
+                ->orderBy('category_id')
+                ->chunk(self::CHUNK_SIZE, function (Collection $categories) use ($contextId, &$fileNamesToMove) {
+                    $imageRecordsToUpdate = [];
+
+                    foreach ($categories as $category) {
+                        $image = $category->image;
+                        if ($image) {
+                            $imageDecoded = json_decode($image, true);
+                            $fileNamesToMove[] = $imageDecoded['name'];
+                            $fileNamesToMove[] = $imageDecoded['thumbnailName'];
+                            // Swap 'name' and 'uploadName' fields
+                            [$imageDecoded['name'], $imageDecoded['uploadName']] = [$imageDecoded['uploadName'], $imageDecoded['name']];
+                            $imageRecordsToUpdate[$category->category_id] = json_encode($imageDecoded);
+                        }
+                    }
+
+                    $this->updateCategoriesImageNameFields($contextId, collect($imageRecordsToUpdate));
+                });
+
+            // Move images
+            $this->moveContextCategoryImagesToPublicFolder($contextId, $fileNamesToMove);
```

Comment:
I think this should be moved inside the chunked part, to avoid having a large `$fileNamesToMove`.